clientapi/routing/aliases.go

```diff
@@ -55,9 +55,23 @@ func GetAliases(
 		visibility = content.HistoryVisibility
 	}
 	if visibility != spec.WorldReadable {
+		fullUserID, err := spec.NewUserID(device.UserID, true)
+		if err != nil {
+			return util.JSONResponse{
+				Code: http.StatusForbidden,
+				JSON: spec.Forbidden("userID doesn't have power level to change visibility"),
+			}
+		}
+		senderID, err := rsAPI.QuerySenderIDForUser(req.Context(), roomID, *fullUserID)
+		if err != nil {
+			return util.JSONResponse{
+				Code: http.StatusForbidden,
+				JSON: spec.Forbidden("userID doesn't have power level to change visibility"),
+			}
+		}
 		queryReq := api.QueryMembershipForUserRequest{
-			RoomID: roomID,
-			UserID: device.UserID,
+			RoomID:   roomID,
+			SenderID: senderID,
```

Comment:
So I can see this becoming a royal pain as we are constantly forced to do the dance of:
 - take device user ID -> make a spec.UserID
 - QuerySenderIDForUser
 - QueryMembershipForUser

Given both query functions use the rsAPI, it would be more ergonomic to just allow `QueryMembershipForUserRequest` to accept a `UserID` (like it did before) and resolve that in one place, rather than expecting the caller to do this.

We can then situationally add a `QueryMembershipForSender` function when we have a sender ID but no user ID.